core/sql/optimizer/BindItemExpr.cpp

```diff
@@ -6173,7 +6173,35 @@ ItemExpr *Assign::bindNode(BindWA *bindWA)
     } // QSTUFF
 
   } // isUserSpecified
+  NABuiltInTypeEnum sourceType =  child(1)->castToItemExpr()->getValueId().getType().getTypeQualifier() ;
+  NABuiltInTypeEnum targetType =  child(0)->castToItemExpr()->getValueId().getType().getTypeQualifier() ;
+  if ((sourceType == NA_CHARACTER_TYPE) && (targetType == NA_LOB_TYPE))
+    {
+      ValueId vid1 = child(1)->castToItemExpr()->getValueId();  
+      // Add a stringToLob node
+      ItemExpr *newChild =  new (bindWA->wHeap()) LOBinsert( vid1.getItemExpr(), NULL, LOBoper::STRING_, FALSE);    
+      newChild->bindNode(bindWA);
+      if (bindWA->errStatus())
+	return boundExpr; 
+      setChild(1, newChild);
+    }
+  /* if ((sourceType == NA_UNKNOWN_TYPE) && (targetType = NA_LOB_TYPE))
```

Comment:
No I will remove the commented out code. I had initilaly tried to do this cast in the binder but it was not the right place. The type coercion needed to be done in the ::doSynthType() method.